civicase.php

```diff
@@ -588,3 +588,16 @@ function _civicase_add_case_category_case_type_entity(array &$entityTypes) {
     ];
   };
 }
+
+/**
+ * Implements hook_civicrm_alterMailParams().
+ */
+function civicase_civicrm_alterMailParams(&$params, $context) {
+  $hooks = [
+    new CRM_Civicase_Hook_alterMailParams_SubjectCaseTypeCategoryProcessor(),
+  ];
+
+  foreach ($hooks as &$hook) {
+    $hook->run($params, $context, CRM_Utils_Request::retrieve('caseid', 'Integer'));
```

Comment:
The case ID, should be fetched within the `CRM_Civicase_Hook_alterMailParams_SubjectCaseTypeCategoryProcessor` class. The reason is that the parameters passed to the hooks here should reflect what is passed to the `civicase_civicrm_alterMailParams` hook. Other parameters needed , each class should have the logic for getting that.